sqldelight-core/src/main/kotlin/com/squareup/sqldelight/core/SqlDelightEnvironment.kt

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (C) 2017 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.sqldelight.core
+
+import com.alecstrong.sqlite.psi.core.SqliteAnnotationHolder
+import com.alecstrong.sqlite.psi.core.SqliteCoreEnvironment
+import com.alecstrong.sqlite.psi.core.psi.SqliteCreateTableStmt
+import com.alecstrong.sqlite.psi.core.psi.SqliteSqlStmt
+import com.intellij.openapi.vfs.StandardFileSystems
+import com.intellij.openapi.vfs.VirtualFileManager
+import com.intellij.psi.PsiDirectory
+import com.intellij.psi.PsiDocumentManager
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFileSystemItem
+import com.intellij.psi.PsiManager
+import com.squareup.sqldelight.core.compiler.SqlDelightCompiler
+import com.squareup.sqldelight.core.lang.SqlDelightFile
+import com.squareup.sqldelight.core.lang.SqlDelightFileType
+import com.squareup.sqldelight.core.lang.SqlDelightParserDefinition
+import com.squareup.sqldelight.core.psi.SqlDelightImportStmt
+import java.io.File
+import java.util.ArrayList
+import java.util.StringTokenizer
+
+/**
+ * Mocks an intellij environment for compiling sqldelight files without an instance of intellij
+ * running.
+ */
+class SqlDelightEnvironment(
+    /**
+     * The sqlite source directories for this environment.
+     */
+    private val sourceFolders: List<File>,
+    /**
+     * The package name to be used for the generated SqlDelightDatabase class.
+     */
+    private val packageName: String,
+    /**
+     * An output directory to place the generated class files.
+     */
+    private val outputDirectory: File
+): SqliteCoreEnvironment(SqlDelightParserDefinition(), SqlDelightFileType, sourceFolders) {
+
+  init {
+    applicationEnvironment.registerApplicationService(SqlDelightFileIndex::class.java, FileIndex())
+  }
+
+  /**
+   * Run the SQLDelight compiler and return the error or success status.
+   */
+  fun generateSqlDelightFiles(): CompilationStatus {
+    val errors = ArrayList<String>()
+    annotate(object : SqliteAnnotationHolder {
+      override fun createErrorAnnotation(element: PsiElement, s: String?) {
+        errors.add(errorMessage(element, s!!))
```

Comment:
That `s!!` looks suspicious, would it be better with a default value?

Reply:
i dont remember why i made the string nullable....it probably shouldnt be, and if it nullable then yea it should have a default